system/lib/html5/emscripten_wget.c

```diff
@@ -0,0 +1,54 @@
+#include <emscripten.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+// Creates all ancestor directories of a given file, if they do not already
+// exist. Returns 0 on success or 1 on error.
+static int mkdirs(const char* file) {
+  char* copy = strdup(file);
+  char* c = copy;
+  while (*c) {
+    // Create any non-trivial (not the root "/") directory.
+    if (*c == '/' && c != copy) {
+      *c = 0;
+      int result = mkdir(copy, 0777);
+      *c = '/';
+      // Continue while we succeed in creating directories or while we see that
+      // they already exist.
+      if (result < 0 && errno != EEXIST) {
+        free(copy);
+        return 1;
+      }
+    }
+    c++;
+  }
+  free(copy);
+  return 0;
+}
+
+void emscripten_wget(const char* url, const char* file) {
+  // Create the ancestor directories.
+  if (mkdirs(file)) {
+    return;
```

Comment:
As a followup we could/should change the return value of this function to return success/failure..

Reply:
That would be a breaking public API change, though?

Reply:
I don't think changing the return type from `void` to `bool` or `int` could break anything could it?

Reply:
Sorry I already did: #19747